### DISCUSSION TITLE
issue: Spread types may only be created from object types. ts(2698)

### DISCUSSION THREAD

**Discussion by curtvict** [react-hook-form/react-hook-form#9544] [Q&A] [answered]
### Version Number

6.15.8

### Codesandbox/Expo snack

https://codesandbox.io/embed/nifty-ritchie-sw2433?fontsize=14&hidenavigation=1&theme=dark

### Steps to reproduce

1. Go to https://codesandbox.io/embed/nifty-ritchie-sw2433?fontsize=14&hidenavigation=1&theme=dark
2. Click "Edit Sandbox" in the left menu
3. Behold...

### Expected behaviour

I'd expect to use the most basic example of React Hook Form as given in the Quick Start guide.

### What browsers are you seeing the problem on?

TypeScript issue, in development environment.

### Relevant log output

`Spread types may only be created from object types. ts(2698)`

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
please use the correct version and upgrade to v7.

**Reply by curtvict** [reply]
Oh dip! Thanks for moving this into a discussion for me. I'm realizing my confusion was around this UI in the documentation:

![image](https://user-images.githubusercontent.com/96080054/206306969-52e4b693-91de-4730-8f4d-536043183dc1.png)

To me that said I was already viewing the `v5/v6` documentation. Clicking on it changes it to "Version 7":

![image](https://user-images.githubusercontent.com/96080054/206307133-1bdc5b4d-ea6b-430c-b9eb-3992aad5e55b.png)

And it shows the same example in Getting Started.

Clicking it again gives me a dropdown which allows me to select "Version 6" which shows the correct documentation. 

I'd suggest just showing the just "Version 7" version of the UI and allowing users to select their desired version from there. That seems more straightforward to me, at least.

**Reply by bluebill1049** [reply]
sorry about that, I haven't got a chance to optimise that big, you need to select the drop-down to change to v6

![image](https://user-images.githubusercontent.com/10513364/206308419-2b75b72b-9e18-4491-9595-8ea6d623a800.png)

I would be great if you can skip v6, as it's pretty deprecated and none maintained. It's 2 years behind as well.

**Reply by curtvict** [reply]
Totally. I'm going to attempt an upgrade now. Thanks for your help!
